pkgdb/src/main.cc

```diff
@@ -193,28 +206,34 @@ main( int argc, char * argv[] )
   setenv( "NIX_SSL_CERT_FILE", NIXPKGS_CACERT_BUNDLE_CRT, 0 );
 
   /* Wrap all execution in an error handler that pretty prints exceptions. */
+  int exit_code = 0;
   try
     {
-      return run( argc, argv );
+      exit_code = run( argc, argv );
     }
   catch ( const flox::FloxException & err )
     {
-      return printAndReturnException( err );
+      exit_code = printAndReturnException( err );
     }
   // TODO: we may want to catch these closer to where they are
   //       originally thrown.
   // TODO: handle IFD build errors.
   catch ( const nix::Error & err )
     {
-      return printAndReturnException(
+      exit_code = printAndReturnException(
         flox::NixException( "running pkgdb subcommand",
                             nix::filterANSIEscapes( err.what(), true ) ) );
     }
   catch ( const std::exception & err )
     {
-      return printAndReturnException(
+      exit_code = printAndReturnException(
         flox::CaughtException( "running pkgdb subcommand", err.what() ) );
     }
+
+#ifndef __APPLE__
+  flox::theSentryReporting.shutdown();
```

Comment:
wouldn't we still need to return/shutdown with the appropriate exit code?

Reply:
While `sentry_close` does return status code, we are currently ignoring it.  I'm not sure what we would do and IF we want to return a bad exit code if only sentry is failing.  I also am not sure we want to expose a warning to the user when that happens.  We could log a debug level message in the `shutdown` method so it's not completely lost?
Otherwise main still exits with the original `exit_code`.

Reply:
nb: "**the**sentryReporting" 🤔 is that a prefix i'm not aware of ^^

Reply:
yeah i guess we could log it, just wanted to ensure we are actually returning with the original error_code, that seems more appropriate than a sentry one